Jellyfin.Server/Migrations/Routines/MigrateLibraryDb.cs

```diff
@@ -74,265 +74,306 @@ public void Perform()
 
-        migrationTotalTime += stopwatch.Elapsed;
-        _logger.LogInformation("Saving MediaStreamInfos entries took {0}.", stopwatch.Elapsed);
-        stopwatch.Restart();
+                    var entity = GetPerson(reader);
+                    if (!peopleCache.TryGetValue(entity.Name, out var personCache))
+                    {
+                        peopleCache[entity.Name] = personCache = (entity, []);
+                    }
 
-        _logger.LogInformation("Start moving People.");
-        const string personsQuery = """
-        SELECT ItemId, Name, Role, PersonType, SortOrder FROM People
-        WHERE EXISTS(SELECT 1 FROM TypedBaseItems WHERE TypedBaseItems.guid = People.ItemId)
-        """;
-        dbContext.Peoples.ExecuteDelete();
-        dbContext.PeopleBaseItemMap.ExecuteDelete();
+                    if (reader.TryGetString(2, out var role))
+                    {
+                    }
 
-        var peopleCache = new Dictionary<string, (People Person, List<PeopleBaseItemMap> Items)>();
-        var baseItemIds = dbContext.BaseItems.Select(b => b.Id).ToHashSet();
+                    int? sortOrder = reader.IsDBNull(4) ? null : reader.GetInt32(4);
 
-        foreach (SqliteDataReader reader in connection.Query(personsQuery))
-        {
-            var itemId = reader.GetGuid(0);
-            if (!baseItemIds.Contains(itemId))
-            {
-                _logger.LogError("Dont save person {0} because its not in use by any BaseItem", reader.GetString(1));
-                continue;
-            }
+                    personCache.Items.Add(new PeopleBaseItemMap()
+                    {
+                        Item = null!,
+                        ItemId = itemId,
+                        People = null!,
+                        PeopleId = personCache.Person.Id,
+                        ListOrder = sortOrder,
+                        SortOrder = sortOrder,
+                        Role = role
+                    });
+                }
 
-            var entity = GetPerson(reader);
-            if (!peopleCache.TryGetValue(entity.Name, out var personCache))
-            {
-                peopleCache[entity.Name] = personCache = (entity, []);
-            }
+                baseItemIds.Clear();
 
-            if (reader.TryGetString(2, out var role))
-            {
-            }
+                foreach (var item in peopleCache)
+                {
+                    operation.JellyfinDbContext.Peoples.Add(item.Value.Person);
+                    operation.JellyfinDbContext.PeopleBaseItemMap.AddRange(item.Value.Items.DistinctBy(e => (e.ItemId, e.PeopleId)));
+                }
 
-            int? sortOrder = reader.IsDBNull(4) ? null : reader.GetInt32(4);
+                peopleCache.Clear();
+            }
 
-            personCache.Items.Add(new PeopleBaseItemMap()
+            using (new TrackedMigrationStep($"saving {operation.JellyfinDbContext.Peoples.Local.Count} People entries", _logger))
```

Comment:
```suggestion
            int entityCount = operation.JellyfinDbContext.Peoples.Local.Count + operation.JellyfinDbContext.PeopleBaseItemMap.Local.Count;
            using (new TrackedMigrationStep($"saving {entityCount} People entries", _logger))
```

I think we should aggregate both